scripts/generateVersion.sh

```diff
@@ -1,14 +1,14 @@
 #!/bin/sh
-cp ./src/version.h.in ./src/version.h -fr
+cp -fr ./src/version.h.in ./src/version.h
 
 HASH=$(git rev-parse HEAD)
 BRANCH=$(git rev-parse --abbrev-ref HEAD)
 MESSAGE=$(git show ${GIT_COMMIT_HASH} | head -n 5 | tail -n 1 | sed -e 's/#//g' -e 's/\"//g')
-DIRTY=$(git diff-index --quiet HEAD -- || echo \"dirty\")
+DIRTY=$(git diff-index --quiet HEAD -- || echo dirty)
 TAG=$(git describe --tags)
 
-awk -i inplace "{sub(/@HASH@/,\"${HASH}\")}1" ./src/version.h
-awk -i inplace "{sub(/@BRANCH@/,\"${BRANCH}\")}1" ./src/version.h
-awk -i inplace "{sub(/@MESSAGE@/,\"${MESSAGE}\")}1" ./src/version.h
-awk -i inplace "{sub(/@DIRTY@/,\"${DIRTY}\")}1" ./src/version.h
-awk -i inplace "{sub(/@TAG@/,\"${TAG}\")}1" ./src/version.h
\ No newline at end of file
+sed -i -e "s/@HASH@/${HASH}/" ./src/version.h
+sed -i -e "s/@BRANCH@/${BRANCH}/" ./src/version.h
+sed -i -e "s/@MESSAGE@/${MESSAGE}/" ./src/version.h
```

Comment:
message can contain weird characters like `/` which will destroy this command. We only sanitize `#` and `"`

Reply:
Fixed by using one of the sanitized characters as a the delimited for `s` sed command.

Also, why `#` is sanitized? It doesn't start a comment in C/C++ and has no impact within quotes in Bourne-style shells.

Reply:
Applied `\"` consistently to avoid surprises from tags or (local) branches containing `/` slash.

Reply:
> why `#` is sanitized?

Ah, it's for GNU awk. For best compat with the previous behavior I'll switch to `#` as delimiter.